src/mapgen/mg_biome.cpp

```diff
@@ -54,6 +54,7 @@ BiomeManager::BiomeManager(Server *server) :
 	b->heat_point      = 0.0;
 	b->humidity_point  = 0.0;
 	b->vertical_blend  = 0;
+	b->weight          = 0.0;
```

Comment:
A default of 0 (should also be just `0` or `0.0f` since it's a float) seems fragile when we divide by this (though it should be overwritten latter). I would prefer a more sensible default of `1` / `1.0f` here. Our style probably prefers the latter.

Reply:
It is not a default, but it is the fallback biome. Used only if no biomes were applicable. The distance computation loop where the weight is used begins at 1, so that should not be executed - we might as well put 1 here. As for float, this notation is consistent with the values above which also do not have `0.0f`.

Reply:
I see. (I still think `0.0f` would be the better / more explicit notation to use (ideally for all float values); `0.0` is a double which will be converted to float implicitly. Not that it would matter here.)

A comment like `b->weight = 0.0; // not used` would be good for clarity though.